src/TLuaInterpreter.cpp

```diff
@@ -11777,13 +11793,81 @@ int TLuaInterpreter::getRoomChar(lua_State* L)
     if (!pR) {
         lua_pushnil(L);
         lua_pushfstring(L, "room with id %d does not exist", id);
-        return 2;
+        return 1;
     } else {
         lua_pushstring(L, pR->mSymbol.toUtf8().constData());
         return 1;
     }
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setRoomNameOffset
+int TLuaInterpreter::setRoomNameOffset(lua_State* L)
+{
+    int id;
+    qreal x, y;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushfstring(L, "setRoomNameOffset: bad argument #1 type (room id as number expected, got %s!)",
+                       luaL_typename(L, 1));
+        return lua_error(L);
+    } else {
+        id = lua_tointeger(L, 1);
+    }
+
+    if (!lua_isnumber(L, 2)) {
+        lua_pushfstring(L, "setRoomNameOffset: bad argument #2 type (room offset X as float expected, got %s!)",
+                       luaL_typename(L, 2));
+        return lua_error(L);
+    } else {
+        x = lua_tonumber(L, 2);
+    }
+
+    if (!lua_isnumber(L, 3)) {
+        lua_pushfstring(L, "setRoomNameOffset: bad argument #3 type (room offset X as float expected, got %s!)",
+                       luaL_typename(L, 3));
+        return lua_error(L);
+    } else {
+        y = lua_tonumber(L, 3);
+    }
+
+    Host& host = getHostFromLua(L);
+    TRoom* pR = host.mpMap->mpRoomDB->getRoom(id);
+    if (!pR) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "room with id %d does not exist", id);
+        return 1;
+    } else {
+        pR->nameOffset.setX(x);
+        pR->nameOffset.setY(y);
+        lua_pushboolean(L, true);
+        return 1;
```

Comment:
In *this* case the `return` has the same value in both branches - so doesn't need to be inside the `if (...) {...} else {...}`!